tests/kafkatest/services/security/security_config.py

```diff
@@ -169,7 +179,14 @@ def setup_sasl(self, node):
         jaas_conf = self.render(jaas_conf_file,  node=node, is_ibm_jdk=is_ibm_jdk,
                                 client_sasl_mechanism=self.client_sasl_mechanism,
                                 enabled_sasl_mechanisms=self.enabled_sasl_mechanisms)
-        node.account.create_file(SecurityConfig.JAAS_CONF_PATH, jaas_conf)
+        if self.static_jaas_conf:
+            node.account.create_file(SecurityConfig.JAAS_CONF_PATH, jaas_conf)
+        else:
+            match = re.search("KafkaClient *{([^}]*)}", jaas_conf.replace("\n", " "))
```

Comment:
I am not sure if it would be better, but it feels like this is the kind of tweaking of templates that should be done in the templates themselves based on forwarding parameters to the template via the class members. In other words, if there are bits that should be swapped out, maybe we should encode those as methods on this class or simply conditionals in the template instead of doing regex search & replace here.